src/Controller/IncentiveImportController.php

```diff
@@ -95,8 +96,25 @@ public function incentiveImportConfirmation(int $id, Request $request, LoggerSer
     /**
      * @Route("/incentive/import/{id}", name="incentiveImportDetails", methods={"GET", "POST"})
      */
-    public function patientStatusImportDetails(int $id, Request $request, IncentiveImportService $incentiveImportService)
+    public function incentiveImportDetails(int $id, Request $request, IncentiveImportService $incentiveImportService)
     {
-        return '';
+        $incentiveImport = $this->em->getRepository(IncentiveImport::class)->findOneBy([
+            'id' => $id,
+            'user' => $this->getUserEntity(),
+            'confirm' => 1
+        ]);
+        if (empty($incentiveImport)) {
+            throw $this->createNotFoundException('Page Not Found!');
+        }
+        //For ajax requests
+        if ($request->isXmlHttpRequest()) {
+            $params = $request->request->all();
+            $incentiveImportRows = $incentiveImport->getIncentiveImportRows()->slice($params['start'], $params['length']);
```

Comment:
Is there a limit on the number of rows that can be imported at once? If not, we may want to consider paginating on the database side, instead of hydrating the entire collection on every request.

Reply:
Yes, there is a 5000 row limit for the CSV import.